### PR TITLE
[Resources] Fix issue preventing deployment of Template Specs located outside of the current subscription context

### PR DESCRIPTION
**Pull request by stuartko**
## Description

Fixes issue preventing deployment of Template Specs located outside of the current subscription context. Issue is described here: https://github.com/Azure/template-specs/issues/40

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@stuartko Do you have any test case to cover the issue you fix?

**Comment by stuartko** [comment]
> @stuartko Do you have any test case to cover the issue you fix?

I do not. My understanding is that as it’s a multi-subscription action, the current test framework is not setup well to handle that. Is that understanding correct? I see many other cross-subscription operations do not have tests for this reason as well.

**Comment by stuartko** [comment]
>  My understanding is that as it’s a multi-subscription action, the current test framework is not setup well to handle that. Is that understanding correct?

@msJinLei : Can you confirm? 

I only see one test (in Az.Accounts) in Az Powershell that is using multiple subscription contexts and it comes with an asterisk that it requires a tenant with two subscriptions that are both relevant to the test. 

Given these limitations, I do not believe a cross-subscription test can be safely implemented for this hotfix.

**Comment by stuartko** [comment]
Adding back "needs-review" label to get reviewer confirmation for previous question in the PR thread. I believe this PR should be good to merge?

**Comment by msJinLei** [comment]
> > My understanding is that as it’s a multi-subscription action, the current test framework is not setup well to handle that. Is that understanding correct?
> 
> @msJinLei : Can you confirm?
> 
> I only see one test (in Az.Accounts) in Az Powershell that is using multiple subscription contexts and it comes with an asterisk that it requires a tenant with two subscriptions that are both relevant to the test.
> 
> Given these limitations, I do not believe a cross-subscription test can be safely implemented for this hotfix.

Sorry for delayed reponse. You can use unit test rather than scenario test and test data can be mocked.

**Comment by stuartko** [comment]
@msJinLei: It was more involved than anticipated due to some shell-only assumptions in the original cmdlet implementation, but I've now completed adding a unit test to cover the missing coverage for the bug being fixed.

Can you review/approve?

**Comment by stuartko** [comment]
Can someone please review? We need to have this change in before Code Complete for the next release.

**Comment by msJinLei** [comment]
> @msJinLei: It was more involved than anticipated due to some shell-only assumptions in the original cmdlet implementation, but I've now completed adding a unit test to cover the missing coverage for the bug being fixed.
> 
> Can you review/approve?

Great！

There's a failed test case in mac and linux

```
2020-11-23T09:51:24.3167640Z   [xUnit.net 00:02:05.19]     Microsoft.Azure.Commands.Resources.Test.NewAzureResourceGroupDeploymentCommandTests.ResolvesDynamicParametersWithCrossSubTemplateSpec [FAIL]
2020-11-23T09:51:24.3353190Z     X Microsoft.Azure.Commands.Resources.Test.NewAzureResourceGroupDeploymentCommandTests.ResolvesDynamicParametersWithCrossSubTemplateSpec [9ms]
2020-11-23T09:51:24.3361432Z EXEC : error Message:  [/home/vsts/work/1/s/build.proj]
2020-11-23T09:51:24.3363762Z      System.IO.FileNotFoundException : Could not find file '/home/vsts/work/1/s/src/Resources/Resources.Test/bin/Debug/netcoreapp2.1/Resources\sampleTemplateFile.json'.
2020-11-23T09:51:24.3364657Z     Stack Trace:
2020-11-23T09:51:24.3365336Z        at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
2020-11-23T09:51:24.3366123Z      at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
2020-11-23T09:51:24.3366963Z      at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
2020-11-23T09:51:24.3367794Z      at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize)
2020-11-23T09:51:24.3368541Z      at System.IO.File.InternalReadAllText(String path, Encoding encoding)
2020-11-23T09:51:24.3369159Z      at System.IO.File.ReadAllText(String path)
```

File: src/Resources/Resources.Test/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCommandTests.cs [hunk 1]
```diff
@@ -271,5 +288,68 @@ public void CreatesNewPSResourceGroupDeploymentWithUserTemplateEmptyRollback()
...
+            const string templateSpecSubscriptionId = "10000000-0000-0000-0000-000000000000";
+            const string templateSpecRGName = "someRG";
+            const string templateSpecName = "myTemplateSpec";
+            const string templateSpecVersion = "v1";
+
+            string templateSpecId = $"/subscriptions/{templateSpecSubscriptionId}/" +
+                $"resourceGroups/{templateSpecRGName}/providers/Microsoft.Resources/" +
+                $"templateSpecs/{templateSpecName }/versions/{templateSpecVersion}";
+
+            var templateContentForTest = File.ReadAllText(templateFile);
```
Comment (msJinLei) [inline src/Resources/Resources.Test/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCommandTests.cs:314]: The path is invalid in linux and mac.

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by stuartko** [comment]
@msJinLei : The issue on Unix based systems is now fixed. We need this change in for the next powershell release, can we have this merged today so we hit the code complete deadline?

**Review by msJinLei — APPROVED** [review]

**Pull request merged**